ReactNativeClient/lib/models/Setting.js

```diff
@@ -334,7 +335,36 @@ class Setting extends BaseModel {
 			'encryption.passwordCache': { value: {}, type: Setting.TYPE_OBJECT, public: false, secure: true },
 			'style.zoom': { value: 100, type: Setting.TYPE_INT, public: true, appTypes: ['desktop'], section: 'appearance', label: () => _('Global zoom percentage'), minimum: 50, maximum: 500, step: 10 },
 			'style.editor.fontSize': { value: 13, type: Setting.TYPE_INT, public: true, appTypes: ['desktop'], section: 'appearance', label: () => _('Editor font size'), minimum: 4, maximum: 50, step: 1 },
-			'style.editor.fontFamily': { value: '', type: Setting.TYPE_STRING, public: true, appTypes: ['desktop'], section: 'appearance', label: () => _('Editor font family'), description: () => _('This must be *monospace* font or it will not work properly. If the font is incorrect or empty, it will default to a generic monospace font.') },
+			'style.editor.fontFamily': {
+				value: '',
+				type: Setting.TYPE_STRING,
+				public: true,
+				appTypes: ['desktop'],
+				section: 'appearance',
+				label: () => _('Editor font family'),
+				description: () =>
+					_('This must be *monospace* font or it will not work properly. If the font ' +
+					'is incorrect or empty, it will default to a generic monospace font.'),
+			},
+			editorFont: {
+				value: Setting.FONT_MENLO,
+				type: Setting.TYPE_STRING,
+				isEnum: true,
+				public: true,
+				label: () => _('Editor font'),
+				appTypes: ['mobile'],
+				section: 'appearance',
+				options: () => {
+					if (mobilePlatform === 'ios') {
+						return {
+							[Setting.FONT_MENLO]: 'Menlo',
```

Comment:
The use of an array as a key is confusing me a bit here. Since we are using an enum, is there any reason why we can't make this setting a TYPE_INT? Then we can still display a string in the setting screen, like it's done in the `theme` setting for example. For me the branch currently throws an exception due to the mix of array/string/int for this setting value (I think because it's initialised to an int, but expects a string, so if we can make a the value an int everywhere it will be easier to see what the issue is).

Reply:
The syntax is a bit funky, but it's not actually setting arrays as keys. It's just syntax for creating a more elegant object, equivalent to what's happening in `theme`, rather than mutating it. A toy example:
```
ReactNativeClient ❯ node
> const x = 'a'
> {[x]: 1}
{ a: 1 }
```

So this...:

```
return {
  [Setting.FONT_MENLO]: 'Menlo',
  [Setting.FONT_COURIER_NEW]: 'Courier New',
  [Setting.FONT_AVENIR]: 'Avenir',
}
```

... is ~equivalent to this:
```
let output = {}
output[Setting.FONT_MENLO] = 'Menlo'
output[Setting.FONT_COURIER_NEW] = 'Courier New'
output[Setting.FONT_AVENIR] = 'Avenir'
return output
```

Reply:
Thanks for clarifying, that makes sense now, and I've learnt something new about JavaScript :-)